manta-trusted-setup/src/groth16/mpc.rs

```diff
@@ -53,6 +56,62 @@ pub type State<P> = ProvingKey<<P as Pairing>::Pairing>;
 /// MPC Proof
 pub type Proof<P> = RatioProof<P>;
 
+/// MPC State Size
+#[cfg_attr(
+    feature = "serde",
+    derive(Deserialize, Serialize),
+    serde(crate = "manta_util::serde", deny_unknown_fields)
+)]
+#[derive(Clone, Copy, Debug, Default, Eq, Hash, PartialEq)]
+pub struct StateSize {
+    /// Size of `gamma_abc_g1` in the [`VerifyingKey`]
+    pub gamma_abc_g1: usize,
+
+    /// Size of the `a_query` in the [`ProvingKey`]
+    ///
+    /// # Note
+    ///
+    /// This is also the size of the `b_g1_query` and the `b_g2_query`.
+    pub a_query: usize,
+
+    /// Size of the `h_query` in the [`ProvingKey`]
+    pub h_query: usize,
+
+    /// Size of the `l_query` in the [`ProvingKey`]
+    pub l_query: usize,
+}
```

Comment:
Two comments:
first, the documentation is unclear, e.g., how do I know what `gamma_abc_g1` is without consulting the docs of `ark-groth16`? 
second, probably we shouldn't name the length of `gamma_abc_g1: usize` the same as the points themselves `gamma_abc_g1: Vec<E1::Affine>`.

Reply:
Yeah, I'd like to replace all of these with universal sizes like:

1. number of constraints
2. number of secret witness variables
3. number of public input variables

But this one for now at least matches the `VerifyingKey` and the entire infrastructure at this level relies on `ark-groth16` anyway. See this issue: https://github.com/Manta-Network/manta-rs/issues/214.